website/docs/r/resource_pool.html.markdown

```diff
@@ -0,0 +1,111 @@
+  default = "cluster1"
+}
+
+data "vsphere_datacenter" "dc" {
+  name = "${var.datacenter}"
+}
+
+data "vsphere_compute_cluster" "compute_cluster" {
+  name          = "${var.cluster}"
+  datacenter_id = "${data.vsphere_datacenter.dc.id}"
+}
+
+resource "vsphere_resource_pool" "resource_pool" {
+  name                = "terraform-compute-cluster-test"
+  root_resource_pool_id = "${data.vsphere_compute_cluster.compute_cluster.id}"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the resource pool.
+* `root_resource_pool_id` - (Required) The [managed object ID][docs-about-morefs]
+  of the root resource pool of the compute resource the resource pool is in.
+* `cpu_share_level` - (Optional) The CPU allocation level. The level is a
+  simplified view of shares. Levels map to a pre-determined set of numeric
+  values for shares. Can be one of `low`, `normal`, `high`, or `custom`. When
+  `low`, `normal`, or `high` are specified values in `cpu_shares` will be
+  ignored.  Default: `normal`
+* `cpu_shares` - (Optional) The number of shares allocated for CPU. Used to
+  determine resource allocation in case of resource contention. If this is set,
+  `cpu_share_level` must be `custom`.
+* `cpu_reservation` - (Optional) Amount of CPU (MHz) that is guaranteed
+  available to the resource pool. Default: `0`
+* `cpu_expandable` - (Optional) Determines if the reservation on a resource
+  pool can grow beyond the specified value if the parent resource pool has
+  unreserved resources. Default: `true`
+* `cpu_limit` - (Optional) The CPU utilization of a resource pool will not exceed
+  this limit, even if there are available resources. Set to `-1` for unlimited.
+  Default: `-1`
+* `memory_share_level` - (Optional) The CPU allocation level. The level is a
+  simplified view of shares. Levels map to a pre-determined set of numeric
+  values for shares. Can be one of `low`, `normal`, `high`, or `custom`. When
+  `low`, `normal`, or `high` are specified values in `memory_shares` will be
+  ignored.  Default: `normal`
+* `memory_shares` - (Optional) The number of shares allocated for CPU. Used to
+  determine resource allocation in case of resource contention. If this is set,
+  `memory_share_level` must be `custom`.
+* `memory_reservation` - (Optional) Amount of CPU (MHz) that is guaranteed
+  available to the resource pool. Default: `0`
+* `memory_expandable` - (Optional) Determines if the reservation on a resource
+  pool can grow beyond the specified value if the parent resource pool has
+  unreserved resources. Default: `true`
+* `memory_limit` - (Optional) The CPU utilization of a resource pool will not exceed
+  this limit, even if there are available resources. Set to `-1` for unlimited.
+  Default: `-1`
+
+[docs-about-morefs]: /docs/providers/vsphere/index.html#use-of-managed-object-references-by-the-vsphere-provider
+
+## Attribute Reference
+
+The following attributes are exported:
```

Comment:
If the only attribute exported is the ID, change this to the paragraph form as seen in places like [here](https://www.terraform.io/docs/providers/vsphere/r/compute_cluster_host_group.html#attribute-reference).